lib/ddtrace/profiling/scheduler.rb

```diff
@@ -0,0 +1,66 @@
+require 'ddtrace/worker'
+require 'ddtrace/workers/polling'
+
+module Datadog
+  module Profiling
+    # Pulls and exports profiling data on an async interval basis
+    class Scheduler < Worker
+      include Workers::Polling
```

Comment:
`Polling` wraps the `perform` function first with an `enabled?` check, then `Async::Thread` behavior, then `IntervalLoop` behavior, in that order. `perform` in this class is what is invoked per interval.